src/stores/store/keys/local/local.go

```diff
@@ -0,0 +1,370 @@
+		secretStore: secretStore,
+		keysDA:      keysDA,
+		logger:      logger,
+	}
+}
+
+func (s *Store) Info(context.Context) (*entities.StoreInfo, error) {
+	return nil, errors.ErrNotImplemented
+}
+
+func (s *Store) Create(ctx context.Context, id string, alg *entities.Algorithm, attr *entities.Attributes) (*entities.Key, error) {
+	return s.createKey(ctx, id, nil, alg, attr)
+}
+
+func (s *Store) Import(ctx context.Context, id string, privKey []byte, alg *entities.Algorithm, attr *entities.Attributes) (*entities.Key, error) {
+	return s.createKey(ctx, id, privKey, alg, attr)
+}
+
+func (s *Store) Get(ctx context.Context, id string) (*entities.Key, error) {
+	logger := s.logger.With("id", id)
+
+	key, err := s.keysDA.Get(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	logger.Debug("key retrieved successfully")
+	return key, nil
+}
+
+func (s *Store) List(ctx context.Context) ([]string, error) {
+	ids := []string{}
+	keysRetrieved, err := s.keysDA.GetAll(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, key := range keysRetrieved {
+		ids = append(ids, key.ID)
+	}
+
+	s.logger.Debug("keys listed successfully")
+	return ids, nil
+}
+
+func (s *Store) Update(ctx context.Context, id string, attr *entities.Attributes) (*entities.Key, error) {
+	logger := s.logger.With("id", id)
+	logger.Debug("updating key")
+
+	key, err := s.keysDA.Get(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	key.Tags = attr.Tags
+
+	err = s.keysDA.Update(ctx, key)
+	if err != nil {
+		return nil, err
+	}
+
+	logger.Info("key updated successfully")
+	return key, nil
+}
+
+func (s *Store) Delete(ctx context.Context, id string) error {
+	logger := s.logger.With("id", id)
+	logger.Debug("deleting key")
+
+	err := s.secretStore.Delete(ctx, id)
+	if err != nil {
+		return err
+	}
+
+	err = s.keysDA.Remove(ctx, id)
+	if err != nil {
+		return err
+	}
+
+	logger.Info("key deleted successfully")
+	return nil
+}
+
+func (s *Store) GetDeleted(ctx context.Context, id string) (*entities.Key, error) {
+	logger := s.logger.With("id", id)
+
+	key, err := s.keysDA.GetDeleted(ctx, id)
+	if err != nil {
+		logger.Error("failed to get deleted key")
+		return nil, err
+	}
+
+	logger.Debug("deleted key retrieved successfully")
+	return key, nil
+}
+
+func (s *Store) ListDeleted(ctx context.Context) ([]string, error) {
+	ids := []string{}
+	keysRetrieved, err := s.keysDA.GetAllDeleted(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, key := range keysRetrieved {
+		ids = append(ids, key.ID)
+	}
+
+	s.logger.Debug("deleted keys listed successfully")
+	return ids, nil
+}
+
+func (s *Store) Undelete(ctx context.Context, id string) error {
+	logger := s.logger.With("id", id)
+	logger.Debug("restoring key")
+
+	key, err := s.keysDA.GetDeleted(ctx, id)
+	if err != nil {
+		return err
+	}
+
+	err = s.secretStore.Undelete(ctx, id)
+	if err != nil {
+		return err
+	}
+
+	err = s.keysDA.Restore(ctx, key)
+	if err != nil {
+		return err
+	}
+
+	err = s.keysDA.Add(ctx, key)
```

Comment:
Why are you adding the key again? As far as I see you already restore it in the store above